Packages/Status/Sources/Status/Row/Subviews/StatusRowTextView.swift

```diff
@@ -6,17 +6,47 @@ struct StatusRowTextView: View {
   @EnvironmentObject private var theme: Theme
 
   let status: AnyStatus
-  let viewModel: StatusRowViewModel
-
+  @ObservedObject var viewModel: StatusRowViewModel
+  
   var body: some View {
-    HStack {
-      EmojiTextApp(status.content, emojis: status.emojis, language: status.language)
-        .font(.scaledBody)
-        .emojiSize(Font.scaledBodyPointSize)
-        .environment(\.openURL, OpenURLAction { url in
-          viewModel.routerPath.handleStatus(status: status, url: url)
-        })
-      Spacer()
+    VStack {
+      HStack {
+        EmojiTextApp(status.content, emojis: status.emojis, language: status.language, lineLimit: viewModel.collapsedLines)
+          .font(.scaledBody)
+          .emojiSize(Font.scaledBodyPointSize)
+          .environment(\.openURL, OpenURLAction { url in
+            viewModel.routerPath.handleStatus(status: status, url: url)
+          })
+        Spacer()
+      }
+      makeCollapseButton()
+    }
+  }
+  
+  @ViewBuilder
+  func makeCollapseButton() -> some View {
+    if viewModel.showCollapseButton && viewModel.isCollapsed {
```

Comment:
Could we move this entire logic to viewmodel?

Reply:
yep, and cleaned up the property getters a little